src/main/kotlin/com/exactpro/th2/ws/client/api/impl/WebSocketClient.kt

```diff
@@ -187,18 +187,18 @@ class WebSocketClient(
         onInfo { "Stopping client" }
 
         socket.runCatching {
-            isRunning = false
-
             if (isOutputClosed) {
+                isRunning = false
                 logger.warn { "Trying to close socket abruptly" }
                 abort()
             } else {
                 logger.info { "Trying to close socket gracefully" }
-
-                handler.runCatching(IHandler::preClose).onFailure {
+                handler.runCatching {
+                    preClose(this@WebSocketClient)
+                }.onFailure {
                     onError(it) { "Failed to handle preClose event" }
                 }
```

Comment:
I think we should move the `preClose` call in a separate block before `socket.runCatching`. And we should leave the `isRunning = false` where it was before.
Currently, we set the `isRunning` to false in two places and I think it is better to leave only one place where we do that

Reply:
As u can see there can be case when socket output is already closed. Is it really good choice to ignore this case?

```
if (isOutputClosed) {
  logger.warn { "Trying to close socket abruptly" }
  abort()
} 
```

I mean. If output is already closed there no point of preClose method for handler due no possibility to send data from preClose method before close event

Reply:
Can we move the `isRunning = false` to the end of the method? Right before `onInfo { "Stopped client" }`

Reply:
I think yes due lock on stop method